crates/astria-conductor/src/block_cache.rs

```diff
@@ -51,6 +51,15 @@ impl<T> BlockCache<T> {
         Some(block)
     }
 
+    pub(crate) fn drop_obsolete(&mut self, latest_height: Height) {
+        self.next_height = std::cmp::max(self.next_height, latest_height);
+        // Splitting the btree always involves an allocation, so only do it if necessary
+        if self.inner.first_key_value().map(|(&height, _)| height) < Some(latest_height) {
+            let only_non_obsolete = self.inner.split_off(&latest_height);
+            self.inner = only_non_obsolete;
+        }
+    }
```

Comment:
is splitting more efficient than clearing everything before `latest_height`?

Reply:
I tried understanding both `split_off` and `remove_entry` and honestly didn't get far with the latter. `BTreeMap` might likely not be the right datastructure for this but I am going to leave `split_off` because it's easier to understand.